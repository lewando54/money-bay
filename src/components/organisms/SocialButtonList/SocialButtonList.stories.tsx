/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'

import SocialButtonList from './SocialButtonList'
import { create } from 'react-test-renderer'

const meta = {
    component: SocialButtonList,
    argTypes: {
        socials: { control: 'array' }
    }
} satisfies Meta<typeof SocialButtonList>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        socials: [
            { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
            { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
            { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
        ],
        testId: 'sociallisttest'
    }
}