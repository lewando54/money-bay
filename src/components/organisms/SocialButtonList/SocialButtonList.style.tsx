import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div: {
        display: 'flex',
        width: '100%',
        justifyContent: 'center',
        gap: 10,
        flexDirection: 'row'
    }
})