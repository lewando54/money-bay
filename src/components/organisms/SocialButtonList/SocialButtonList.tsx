import React from 'react'
import { View } from 'react-native'

import { SocialButton } from '@atoms'
import { TiconProp } from 'src/components/atoms/SocialButton/SocialButton'

import socialButtonListStyle from './SocialButtonList.style'

export type TSocialsArray = Array<{ id: number, color: string, icon: TiconProp, href: URL['href'], testId?: string }>
interface ISocialButtonList {
  socials: TSocialsArray
  testId?: string
}

function SocialButtonList ({ socials, testId }: ISocialButtonList): React.ReactElement {
    const socialItems = socials.map(social =>
        <SocialButton size={28} key={social.id} color={social.color} href={social.href} icon={social.icon} testId={testId + social.id.toString()}/>
    )

    return (
        <View testID={testId} style={socialButtonListStyle.div}>
            {socialItems}
        </View>
    )
}

export default SocialButtonList
