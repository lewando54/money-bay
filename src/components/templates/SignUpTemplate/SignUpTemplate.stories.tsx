/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { useState } from 'react'
import { Meta,StoryObj } from '@storybook/react-native'
import { TSocialsArray } from 'src/components/organisms/SocialButtonList/SocialButtonList'

import SignUpTemplate from './SignUpTemplate'

interface IStatefulSignUpMockupProps {
    emailValue?: string
    passwordValue?: string
    passwordConfirmValue?: string
    nameValue?: string
    onSubmit?: () => void
    onSignInPress?: () => void
    socials?: TSocialsArray
}

function StatefulSignUp({
    emailValue,
    passwordValue,
    passwordConfirmValue,
    nameValue,
    onSubmit,
    onSignInPress,
    socials,
}: IStatefulSignUpMockupProps){
    const [name, setName] = useState(nameValue)
    const [email, setEmail] = useState(emailValue)
    const [password, setPassword] = useState(passwordValue)
    const [confirmPassword, setConfirmPassword] = useState(passwordConfirmValue)

    return (
        <SignUpTemplate
            nameValue={name}
            onNameChange={(newValue) => setName(() => newValue)}
            emailValue={email}
            onEmailChange={(newValue) => setEmail(() => newValue)}
            passwordValue={password}
            onPasswordChange={(newValue) => setPassword(() => newValue)}
            passwordConfirmValue={confirmPassword}
            onPasswordConfirmChange={(newValue) => setConfirmPassword(() => newValue)}
            onSignInPress={onSignInPress}
            onSubmit={onSubmit}
            socials={socials}
        />
    )
}

const meta = {
    component: StatefulSignUp,
    argTypes: {
        emailValue: {control: 'text'},
        passwordValue: {control: 'text'},
        passwordConfirmValue: {control: 'text'},
        nameValue: {control: 'text'},
        onSubmit: {action: 'Submitted'},
        onSignInPress: {action: 'SignIn Pressed'},
        socials: { control: 'object' }
    }
} satisfies Meta<typeof StatefulSignUp>

export default meta
type Story = StoryObj<typeof StatefulSignUp>

export const Default: Story = {
    args: {
        socials: [
            { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
            { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
            { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
        ],
    }
}
