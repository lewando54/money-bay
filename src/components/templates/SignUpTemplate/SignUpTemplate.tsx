import React from 'react'
import { Text,View } from 'react-native'

import { Button, Input, Link } from '@atoms'
import { EmailInput, PasswordInput } from '@molecules'
import { SocialButtonList } from '@organisms'
import { TSocialsArray } from 'src/components/organisms/SocialButtonList/SocialButtonList'

import signUpTemplateStyle from './SignUpTemplate.style'

interface ISignUpTemplateProps {
    emailValue?: string
    passwordValue?: string
    passwordConfirmValue?: string
    nameValue?: string
    onNameChange?: (newValue: string) => void
    onEmailChange?: (newValue: string) => void
    onPasswordChange?: (newValue: string) => void
    onPasswordConfirmChange?: (newValue: string) => void
    onSubmit?: () => void
    onSignInPress?: () => void
    socials: TSocialsArray
    testID?: string
}

export default function SignUpTemplate({
    emailValue,
    passwordValue,
    passwordConfirmValue,
    nameValue,
    onNameChange,
    onEmailChange,
    onPasswordChange,
    onPasswordConfirmChange,
    onSubmit,
    onSignInPress,
    socials,
    testID
}: ISignUpTemplateProps){
    return (
        <View style={signUpTemplateStyle.mainContainer} testID={testID}>
            <Text style={signUpTemplateStyle.heading}>Sign up!</Text>
            <Input value={nameValue} onChange={onNameChange} placeholder='Enter your name'/>
            <EmailInput value={emailValue} onChange={onEmailChange} placeholder='Enter your email'/>
            <PasswordInput value={passwordValue} onChange={onPasswordChange} placeholder='Enter your password'/>
            <PasswordInput value={passwordConfirmValue} onChange={onPasswordConfirmChange} placeholder='Confirm your password'/>
            <Button onClick={onSubmit}>Sign up</Button>
            <View style={signUpTemplateStyle.registerTextContainer}>
                <Text style={signUpTemplateStyle.rememberText}>Already have an account?</Text>
                <Link onPress={onSignInPress}>Sign in.</Link>
            </View>
            <SocialButtonList socials={socials}/>
        </View>
    )
}
