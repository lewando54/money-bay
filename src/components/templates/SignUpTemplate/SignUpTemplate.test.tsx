import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import SignUpTemplate from './SignUpTemplate'
import { TSocialsArray } from '../../organisms/SocialButtonList/SocialButtonList'
describe('SignUpTemplate', () => {
    const socials: TSocialsArray = [
        { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
        { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
        { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
    ]

    it('should render correctly', () => {
        const tree = renderer.create(<SignUpTemplate socials={socials}/>)
        expect(tree).toMatchSnapshot()
    })
})
