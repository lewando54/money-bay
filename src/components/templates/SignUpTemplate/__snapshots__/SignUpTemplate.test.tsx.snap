// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpTemplate should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "gap": 25,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
        {
          "fontFamily": "MulishBold",
          "fontSize": 44,
          "textAlign": "center",
        },
      ]
    }
  >
    Sign up!
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        [
          {
            "alignItems": "center",
            "borderColor": "white",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 14,
          },
          undefined,
          false,
        ],
      ]
    }
    testID="universal-container"
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Enter your name"
      placeholderTextColor="#868698"
      secureTextEntry={false}
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "marginRight": 10,
            "maxWidth": "93%",
            "width": "100%",
          },
          undefined,
          {
            "maxWidth": "100%",
          },
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={10}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "none",
        }
      }
      testID="input-icon"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        [
          {
            "alignItems": "center",
            "borderColor": "white",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 14,
          },
          undefined,
          false,
        ],
      ]
    }
    testID="universal-container"
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Enter your email"
      placeholderTextColor="#868698"
      secureTextEntry={false}
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "marginRight": 10,
            "maxWidth": "93%",
            "width": "100%",
          },
          undefined,
          {
            "maxWidth": "100%",
          },
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={10}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "none",
        }
      }
      testID="input-icon"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        [
          {
            "alignItems": "center",
            "borderColor": "white",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 14,
          },
          undefined,
          false,
        ],
      ]
    }
    testID="universal-container"
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Enter your password"
      placeholderTextColor="#868698"
      secureTextEntry={true}
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "marginRight": 10,
            "maxWidth": "93%",
            "width": "100%",
          },
          undefined,
          {
            "maxWidth": "100%",
          },
        ]
      }
      testID="passwordtest"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={10}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "none",
        }
      }
      testID="input-icon"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        [
          {
            "alignItems": "center",
            "borderColor": "white",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 14,
          },
          undefined,
          false,
        ],
      ]
    }
    testID="universal-container"
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Confirm your password"
      placeholderTextColor="#868698"
      secureTextEntry={true}
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "marginRight": 10,
            "maxWidth": "93%",
            "width": "100%",
          },
          undefined,
          {
            "maxWidth": "100%",
          },
        ]
      }
      testID="passwordtest"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={10}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "none",
        }
      }
      testID="input-icon"
    >
      <Text />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4288142079,
            4279654301,
          ],
          "endPoint": [
            1,
            0.5,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0.5,
          ],
        }
      }
      style={
        [
          {
            "borderRadius": 10,
            "padding": 15,
          },
          false,
          false,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 14,
            },
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "color": "white",
              },
              false,
              false,
            ],
          ]
        }
      >
        Sign up
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
        ]
      }
    >
      Already have an account?
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "#FF5887",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            undefined,
          ],
        ]
      }
    >
      Sign in.
    </Text>
  </View>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "gap": 10,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="undefined0"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
            {
              "backgroundColor": "#3B5999",
              "borderRadius": 49,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="pressable-social-button"
      >
        <Text />
      </View>
    </View>
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="undefined1"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
            {
              "backgroundColor": "#55ACEE",
              "borderRadius": 49,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="pressable-social-button"
      >
        <Text />
      </View>
    </View>
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="undefined2"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
            {
              "backgroundColor": "#DD4B39",
              "borderRadius": 49,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="pressable-social-button"
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
